### PR TITLE
[Merged by Bors] - feat: Locally linear graphs

### PR DESCRIPTION
**Pull request by YaelDillies**
Define predicates for a graph to have edge-disjoint triangles and to be locally linear (edge-disjoint triangles and each edge belongs to a triangle).

---

Rehash of https://github.com/leanprover-community/mathlib/pull/19201

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by TFBloom — APPROVED** [review]
Mathematically this is fine - the definitions and statements are all correct and agree with how things are done in the literature.

File: Mathlib/Combinatorics/SimpleGraph/Triangle/Basic.lean [hunk 1]
```diff
@@ -28,23 +29,163 @@ This module defines and proves properties about triangles in simple graphs.
...
+    rintro hG a b hab
+    simp only [Sym2.isDiag_iff_proj_eq] at hab
+    rw [this _ _ (Sym2.mk_isDiag_iff.not.2 hab)]
+    rintro _ ⟨hab, c, hac, hbc, rfl⟩ _ ⟨-, d, had, hbd, rfl⟩
+    refine hG.eq ?_ ?_ (Set.Nontrivial.not_subsingleton ⟨a, ?_, b, ?_, hab.ne⟩) <;>
+      simp [is3Clique_triple_iff, *]
+  · simp only [EdgeDisjointTriangles, is3Clique_iff, Set.Pairwise, mem_cliqueSet_iff, Ne,
+      forall_exists_index, and_imp, ← Set.not_nontrivial_iff (s := _ ∩ _), not_imp_not,
+      Set.Nontrivial, Set.mem_inter_iff, mem_coe]
+    rintro hG _ a b c hab hac hbc rfl _ d e f hde hdf hef rfl g hg₁ hg₂ h hh₁ hh₂ hgh
```
Comment (Ruben-VandeVelde) [inline Mathlib/Combinatorics/SimpleGraph/Triangle/Basic.lean:129]: 🤯

**Review by Ruben-VandeVelde — APPROVED** [review]
maintainer merge

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**